cmd/contour/servecontext_test.go

```diff
@@ -191,9 +200,167 @@ leaderelection:
 	}
 }
 
+// testdata for this test case can be re-generated by running:
+// make gencerts
+// cp certs/*.pem cmd/contour/testdata/X/
+func TestServeContextCertificateHandling(t *testing.T) {
+	tests := map[string]struct {
+		serverCredentialsDir string
+		clientCredentialsDir string
+		expectedServerCert   string
+		expectError          bool
+	}{
+		"successful TLS connection established": {
+			serverCredentialsDir: "testdata/1",
+			clientCredentialsDir: "testdata/1",
+			expectedServerCert:   "testdata/1/contourcert.pem",
+			expectError:          false,
+		},
+		"rotating server credentials returns new server cert": {
+			serverCredentialsDir: "testdata/2",
+			clientCredentialsDir: "testdata/2",
+			expectedServerCert:   "testdata/2/contourcert.pem",
+			expectError:          false,
+		},
+		"fail to connect with bad client certificate": {
+			serverCredentialsDir: "testdata/2",
+			clientCredentialsDir: "testdata/1",
+			expectedServerCert:   "testdata/2/contourcert.pem",
+			expectError:          true,
+		},
+	}
+
+	// create temporary directory to store certificates and key for the server
+	configDir, err := ioutil.TempDir("", "contour-testdata-")
+	checkErr(t, err)
```

Comment:
In this case, `checkErr` should call `t.Fatalf`. It looks like all the other uses in this file should be fatal too, so we can rename it to `checkFatalErr` or something, and update it.

Reply:
I renamed `checkErr()` to `checkFatalErr()` and changed it to call `t.Fatal`.